Add some basic testing

cdrage opened the pull request:
This adds some basic testing that will:

- Grep / sed what version is currently being used in master
- Retrieve the current master binary
- Test against the entire e2e testing suite with the built image in this
repo.

cdrage commented:
Hey @kadel I finally got tests to pass.

Mind doing a quick review?

geoand commented on scripts/testing.sh:
```diff
@@ -0,0 +1,24 @@
+sudo chmod +x /usr/bin/odo
+
+# Go get Odo
+go get github.com/redhat-developer/odo
```
I might be missing something completely here, but why is this needed?

cdrage replied:
We require this in order to run the e2e integration tests from the $GOPATH.

We are forced to use TravisCI under "bash" mode and to install / get Odo manually rather than "Go" mode since this repo isn't exactly a Go project, but rather a Docker image repo.

If that makes any sense!

geoand replied:
Ah yes, I see :) 
Have had similar issues with CircleCI.

One thing I am not clear about though (being a go n00b), when we do `go get github.com/redhat-developer/odo` do we pull master or the latest released version?

geoand replied:
Does the above correspond to the source code that produces the binary at `https://dl.bintray.com/odo/odo/latest/linux-amd64/odo`?

cdrage replied:
Yup. `go get` gets the latest source code.

I use bintray to get the binary so we don't have to spend time building the binary each-time within this test code, but now that you bring it up, I may just do that instead..

geoand replied:
I understand the rationale, what you have done makes perfect sense :)
I just want to be sure that the source and binary match :P

geoand commented:
This is very cool! :+1:

cdrage committed:
Add some basic testing

This adds some basic testing that will:

- Grep / sed what version is currently being used in master
- Retrieve the current master binary
- Test against the entire e2e testing suite with the built image in this
repo.

cdrage commented:
I don't see any glaring issues at the moment, but I'm going to merge this in for now :+1:
